megamek/src/megamek/common/Entity.java

```diff
@@ -9761,7 +9761,12 @@ public int getKillerId() {
     public boolean isEligibleFor(IGame.Phase phase) {
         // only deploy in deployment phase
         if ((phase == IGame.Phase.PHASE_DEPLOYMENT) == isDeployed()) {
-            return false;
+            if ((isDeployed() == false) && isEligibleForTargetingPhase() && game.getOptions().booleanOption(
+                    OptionsConstants.ADVCOMBAT_ON_MAP_PREDESIGNATE)) {
+                    MegaMek.getLogger().info("Artillery Units Present and Advanced PreDesignate option enabled");
```

Comment:
I'd recommend splitting the lines based on the &&'s, as that's easier to read 
```suggestion
            if ((isDeployed() == false) && isEligibleForTargetingPhase()
                    && game.getOptions().booleanOption(OptionsConstants.ADVCOMBAT_ON_MAP_PREDESIGNATE)) {
                    MegaMek.getLogger().info("Artillery Units Present and Advanced PreDesignate option enabled");
```